.circleci/config.yml

```diff
@@ -24,6 +24,50 @@ integrationDefaults: &integrationDefaults
     TAG: dontpush
     GOPATH: /go
 
+# Common procedure to initialize working directory
+initWorkingDir: &initWorkingDir
+  type: shell
+  name: Initialize Working Directory
+  pwd: /
+  command: |
+    sudo mkdir -p /go/src/istio.io/istio
+    sudo chown -R circleci /go
+    mkdir -p /go/out
+    mkdir -p /home/circleci/logs
+
+recordZeroExitCodeIfTestPassed: &recordZeroExitCodeIfTestPassed
+  run:
+    when: on_success
+    name: Record zero exit code as test passed
+    command: echo 0 > exit_code
+
+recordNonzeroExitCodeIfTestFailed: &recordNonzeroExitCodeIfTestFailed
+  run:
+    when: on_fail
+    name: Record nonzero exit code as test failed
+    command: echo 1 > exit_code
+
+markJobStartsOnGCS: &markJobStartsOnGCS
+  run:
+    when: always
+    command: bin/ci2gubernator.sh --job_starts
+
+markJobFinishesOnGCS: &markJobFinishesOnGCS
+  run:
+    when: always
+    command: bin/ci2gubernator.sh --exit_code=$(cat exit_code) --junit_xml=/tmp/junit.xml
```

Comment:
junit.xml is used by circleci for its internal reporting - and should be generated under out/tetss (see store_test_results)

Reply:
Done